src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationCodeFixProvider.cs

```diff
@@ -177,6 +178,32 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             editor.ReplaceNode(identifier, declarationExpression);
         }
 
+        public static TypeSyntax GenerateTypeSyntaxOrVar(
+           ITypeSymbol symbol, OptionSet options)
+        {
+            var useVar = IsVarDesired(symbol, options);
+
+            // Note: we cannot use ".GenerateTypeSyntax()" only here.  that's because we're
+            // actually creating a DeclarationExpression and currently the Simplifier cannot
+            // analyze those due to limitations between how it uses Speculative SemanticModels
+            // and how those don't handle new declarations well.
```

Comment:
note: this is a darn shame.  but i don't have a good way to address this.  it would take some pretty intense plumbing in the simplifier to allow it to make real semantic-models, not just speculative ones (which could have some big perf implications).  Alternatively, if the compiler improved and was better about declaration expressions in semantic models, that could help as well. Taht issue has been reported over here as well: https://github.com/dotnet/roslyn/issues/20724

Reply:
So, for now, because this feature is literally the "create an out-var" feature, it needs to special case how it does 'var' itself, and cannot depend on the simplifier.